### PR TITLE
[WIP] HF Hub Integration

### PR DESCRIPTION
**Pull request by natolambert**
Working towards closing #169

Things to do (roughly):
* Verify base functionality,
* Colab example for loading / saving / visualizing models,
* Upload pretrained models to hub from @luisenp.

### CODE REVIEW

File: mbrl/util/huggingface.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    :param local_path: repository directory
+    :param logdir: logdir directory
+    """
+    if logdir.exists() and logdir.is_dir():
+        # Add the logdir to the repository under new dir called logs
+        repo_logdir = local_path / "logs"
+
+        # Delete current logs if they exist
+        if repo_logdir.exists():
+            shutil.rmtree(repo_logdir)
```
Comment (Wauplin) [inline mbrl/util/huggingface.py:164]: Here `repo_logdir` is a tmp dir (at least when called from `package_to_hub`.
So deleting it locally will not delete it on the Hub. A possibility is to use `delete_folder` to delete the logs folder on the Hub before uploading the new one.

File: mbrl/util/huggingface.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+    """
+
+    msg = (
+        "This function will save your agent, "
+        "create a model card and push everything to the hub. "
+        "This is a work in progress: if you encounter a bug, please open an issue."
+    )
+    print(msg)
+
+    repo_url = HfApi().create_repo(
```
Comment (Wauplin) [inline mbrl/util/huggingface.py:214]: (nit) for consistency with `upload_folder` calls, I would import `create_repo` directly from `huggingface_hub`.
If not called with any parameter, using `HfApi().` is not needed.

File: mbrl/util/huggingface.py [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+):
+    """
+    Upload a model to Hugging Face Hub.
+    :param repo_id: repo_id: id of the model repository from the Hugging Face Hub
+    :param filename: name of the model zip or mp4 file from the repository
+    :param commit_message: commit message
+    :param token
+    """
+
+    repo_url = HfApi().create_repo(
```
Comment (Wauplin) [inline mbrl/util/huggingface.py:300]: (nit) same as above about `HfApi`

**Review by Wauplin — COMMENTED** [review]
Hi @natolambert , really nice implementation ! :)
I left a comment about the `log` directory. I think it should be deleted from the Hub before uploading otherwise you can end up with a messy foler. Other than that, looks good :+1:

**Comment by natolambert** [comment]
WIP [Colab](https://colab.research.google.com/drive/158ZJ0nB0aJkml_yVpoUHSqQlPE-glK1G?usp=sharing) to showcase this workflow.

**Review by luisenp — CHANGES_REQUESTED** [review]
Thanks @natolambert! I left a bunch of comments, but mostly pedantic stuff about formatting and docstrings. I'm not too familiar with HF Hub, so most of the rest LGTM.  However, I noticed that `package_hub` only receives a `Model`, even though we also have algorithms that require a trained `Agent` (e.g., MBPO, and eventually Dreamer). This seems like a potential issue?

Also, is the expectation that should be able to download saved model/agents from the hub and run them on our example scripts?

**Pull request closed without merging**